src/main/java/io/vertx/ext/mongo/IndexOptions.java

```diff
@@ -546,55 +548,36 @@ public IndexOptions partialFilterExpression(JsonObject partialFilterExpression)
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
-
-    IndexOptions options = (IndexOptions) o;
-
-    if (background != options.background) return false;
-    if (unique != options.unique) return false;
-    if (name != null ? !name.equals(options.name) : options.name != null) return false;
-    if (sparse != options.sparse) return false;
-    if (expireAfterSeconds != null ? !expireAfterSeconds.equals(options.expireAfterSeconds) : options.expireAfterSeconds != null)
-      return false;
-    if (version != null ? !version.equals(options.version) : options.version != null) return false;
-    if (weights != null ? !weights.equals(options.weights) : options.weights != null) return false;
-    if (defaultLanguage != null ? !defaultLanguage.equals(options.defaultLanguage) : options.defaultLanguage != null)
-      return false;
-    if (languageOverride != null ? !languageOverride.equals(options.languageOverride) : options.languageOverride != null)
-      return false;
-    if (textVersion != null ? !textVersion.equals(options.textVersion) : options.textVersion != null) return false;
-    if (sphereVersion != null ? !sphereVersion.equals(options.sphereVersion) : options.sphereVersion != null)
-      return false;
-    if (bits != null ? !bits.equals(options.bits) : options.bits != null) return false;
-    if (min != null ? !min.equals(options.min) : options.min != null) return false;
-    if (max != null ? !max.equals(options.max) : options.max != null) return false;
-    if (bucketSize != null ? !bucketSize.equals(options.bucketSize) : options.bucketSize != null) return false;
-    if (storageEngine != null ? !storageEngine.equals(options.storageEngine) : options.storageEngine != null)
-      return false;
-    if (partialFilterExpression != null ? !partialFilterExpression.equals(options.partialFilterExpression) : options.partialFilterExpression != null)
-      return false;
-
-    return true;
+    IndexOptions that = (IndexOptions) o;
+    return isBackground() == that.isBackground() && isUnique() == that.isUnique() && isSparse() == that.isSparse() && Objects.equals(getName(), that.getName()) && Objects.equals(expireAfterSeconds, that.expireAfterSeconds) && Objects.equals(getVersion(), that.getVersion()) && Objects.equals(getWeights(), that.getWeights()) && Objects.equals(getDefaultLanguage(), that.getDefaultLanguage()) && Objects.equals(getLanguageOverride(), that.getLanguageOverride()) && Objects.equals(getTextVersion(), that.getTextVersion()) && Objects.equals(getSphereVersion(), that.getSphereVersion()) && Objects.equals(getBits(), that.getBits()) && Objects.equals(getMin(), that.getMin()) && Objects.equals(getMax(), that.getMax()) && Objects.equals(getBucketSize(), that.getBucketSize()) && Objects.equals(getStorageEngine(), that.getStorageEngine()) && Objects.equals(getPartialFilterExpression(), that.getPartialFilterExpression()) && Objects.equals(getCollation(), that.getCollation());
```

Comment:
There are several changes like this one (reformatting or fields replaced with getter invocations). This makes the PR big and thus important changes are more likely to be missed.

Please try to avoid doing this in future contributions.

Reply:
I may have missed the opportunity to describe these changes:
Since equality/hashcode wasn't implemented properly with my last PR in 4.2.3 I had to change the respective methods in various data objects. As this is an error prone and rather time-consuming task, I decided to let my IDE generate the code. It uses the java way of null-safe comparing objects (which imho is the better way in those cases).
Yes, the PR seems to get bulky, but the effective changes should (at least with this description) easy to grasp.